diff --git a/vendor/curl-sys/build.rs b/vendor/curl-sys/build.rs
index 24475cc..3249440 100644
--- a/vendor/curl-sys/build.rs
+++ b/vendor/curl-sys/build.rs
@@ -111,6 +111,8 @@ fn main() {
         .include("curl/lib")
         .include("curl/include")
         .define("BUILDING_LIBCURL", None)
+        .define("CURL_CA_BUNDLE", "\"/etc/ssl/cert.pem\"")
+        .define("CURL_CA_PATH", "\"/etc/ssl/certs\"")
         .define("CURL_DISABLE_DICT", None)
         .define("CURL_DISABLE_GOPHER", None)
         .define("CURL_DISABLE_IMAP", None)
