diff --git a/vendor/openssl-sys/build/main.rs b/vendor/openssl-sys/build/main.rs
index 49f26d2b7..27deb1f36 100644
--- a/vendor/openssl-sys/build/main.rs
+++ b/vendor/openssl-sys/build/main.rs
@@ -221,6 +221,8 @@ See rust-openssl README for more information:
             (3, 1, 0) => ('3', '1', '0'),
             (3, 1, _) => ('3', '1', 'x'),
             (3, 2, 0) => ('3', '2', '0'),
+            (3, 3, 0) => ('3', '3', '0'),
+            (3, 3, _) => ('3', '3', 'x'),
             _ => version_error(),
         };
 
