diff --git a/tools/profiler/core/platform-linux-android.cpp b/tools/profiler/core/platform-linux-android.cpp
index 07857f2..92e693c 100644
--- a/tools/profiler/core/platform-linux-android.cpp
+++ b/tools/profiler/core/platform-linux-android.cpp
@@ -72,12 +72,7 @@ using namespace mozilla;
 int profiler_current_process_id() { return getpid(); }

 int profiler_current_thread_id() {
-  // glibc doesn't provide a wrapper for gettid().
-#if defined(__GLIBC__)
   return static_cast<int>(static_cast<pid_t>(syscall(SYS_gettid)));
-#else
-  return static_cast<int>(gettid());
-#endif
 }

 void* GetStackTop(void* aGuess) { return aGuess; }
