diff --git a/accessible/ipc/extension/other/moz.build b/accessible/ipc/extension/other/moz.build
index 635bb24fff..983dc974d7 100644
--- a/accessible/ipc/extension/other/moz.build
+++ b/accessible/ipc/extension/other/moz.build
@@ -9,7 +9,7 @@

 IPDL_SOURCES += ['PDocAccessiblePlatformExt.ipdl']

-if CONFIG['ACCESSIBILITY']:
+if 1:
     EXPORTS.mozilla.a11y += [
         'DocAccessiblePlatformExtChild.h',
         'DocAccessiblePlatformExtParent.h',
