diff --git a/layout/generic/WritingModes.h b/layout/generic/WritingModes.h
index 1a8d08b191..0671388902 100644
--- a/layout/generic/WritingModes.h
+++ b/layout/generic/WritingModes.h
@@ -518,7 +518,8 @@ class WritingMode {
    */
   void SetDirectionFromBidiLevel(uint8_t level) {
     if (IS_LEVEL_RTL(level) == IsBidiLTR()) {
-      mWritingMode ^= StyleWritingMode_RTL | StyleWritingMode_INLINE_REVERSED;
+      mWritingMode.bits ^= static_cast<uint8_t>(
+          (StyleWritingMode_RTL | StyleWritingMode_INLINE_REVERSED).bits);
     }
   }
 
diff --git a/servo/ports/geckolib/cbindgen.toml b/servo/ports/geckolib/cbindgen.toml
index bed86498ea..d675ba393d 100644
--- a/servo/ports/geckolib/cbindgen.toml
+++ b/servo/ports/geckolib/cbindgen.toml
@@ -617,12 +617,3 @@ renaming_overrides_prefixing = true
   inline nsRect ToLayoutRect(nscoord aAutoSize = NS_MAXSIZE) const;
 """
 
-"WritingMode" = """
-  StyleWritingMode operator^(const StyleWritingMode& other) const {
-    return {static_cast<decltype(bits)>(this->bits ^ other.bits)};
-  }
-  StyleWritingMode& operator^=(const StyleWritingMode& other) {
-    *this = (*this ^ other);
-    return *this;
-  }
-"""
