diff --git a/Lib/ensurepip/__init__.py b/Lib/ensurepip/__init__.py
index f28ab11..6627189 100644
--- a/Lib/ensurepip/__init__.py
+++ b/Lib/ensurepip/__init__.py
@@ -180,7 +180,7 @@ def _bootstrap(*, root=None, upgrade=False, user=False,
             additional_paths.append(filename)
 
         # Construct the arguments to be passed to the pip command
-        args = ["install", "--no-cache-dir", "--no-index", "--find-links", tmpdir]
+        args = ["install", "-I", "--no-cache-dir", "--no-index", "--find-links", tmpdir]
         if root:
             args += ["--root", root]
         if upgrade:
