diff --git chromium-75.0.3770.100/base/native_library_posix.cc chromium-75.0.3770.100/base/native_library_posix.cc
index 7a5c6f5..6e8a5fa 100644
--- chromium-75.0.3770.100/base/native_library_posix.cc
+++ chromium-75.0.3770.100/base/native_library_posix.cc
@@ -34,7 +34,7 @@ NativeLibrary LoadNativeLibraryWithOptions(const FilePath& library_path,
   // further investigation, as it might vary across versions. Crash here to
   // warn developers that they're trying to rely on uncertain behavior.
   CHECK(!options.prefer_own_symbols);
-#else
+#elif defined(RTLD_DEEPBIND)
   if (options.prefer_own_symbols)
     flags |= RTLD_DEEPBIND;
 #endif
