diff --git chromium-75.0.3770.100/net/dns/dns_config_service_posix.cc chromium-75.0.3770.100/net/dns/dns_config_service_posix.cc
index 48ee925..3bf8756 100644
--- chromium-75.0.3770.100/net/dns/dns_config_service_posix.cc
+++ chromium-75.0.3770.100/net/dns/dns_config_service_posix.cc
@@ -150,7 +150,7 @@ ConfigParsePosixResult ReadDnsConfig(DnsConfig* dns_config) {
 #if !defined(OS_ANDROID)
   ConfigParsePosixResult result;
 // TODO(fuchsia): Use res_ninit() when it's implemented on Fuchsia.
-#if defined(OS_OPENBSD) || defined(OS_FUCHSIA)
+#if defined(OS_OPENBSD) || defined(OS_FUCHSIA) || (defined(OS_LINUX) && !defined(__GLIBC__))
   // Note: res_ninit in glibc always returns 0 and sets RES_INIT.
   // res_init behaves the same way.
   memset(&_res, 0, sizeof(_res));
@@ -173,7 +173,7 @@ ConfigParsePosixResult ReadDnsConfig(DnsConfig* dns_config) {
 #else
   res_nclose(&res);
 #endif  // defined(OS_MACOSX) || defined(OS_FREEBSD)
-#endif  // defined(OS_OPENBSD)
+#endif  // defined(OS_OPENBSD) || defined(OS_FUCHSIA) || (defined(OS_LINUX) && !defined(__GLIBC__))
 
 #if defined(OS_MACOSX) && !defined(OS_IOS)
   ConfigParsePosixResult error = DnsConfigWatcher::CheckDnsConfig();
diff --git chromium-75.0.3770.100/net/dns/dns_reloader.cc chromium-75.0.3770.100/net/dns/dns_reloader.cc
index 03e248c..6135fcc 100644
--- chromium-75.0.3770.100/net/dns/dns_reloader.cc
+++ chromium-75.0.3770.100/net/dns/dns_reloader.cc
@@ -4,8 +4,7 @@
 
 #include "net/dns/dns_reloader.h"
 
-#if defined(OS_POSIX) && !defined(OS_MACOSX) && !defined(OS_OPENBSD) && \
-    !defined(OS_ANDROID) && !defined(OS_FUCHSIA)
+#if defined(__GLIBC__)
 
 #include <resolv.h>
 
@@ -108,5 +107,4 @@ void DnsReloaderMaybeReload() {
 
 }  // namespace net
 
-#endif  // defined(OS_POSIX) && !defined(OS_MACOSX) && !defined(OS_OPENBSD) &&
-        // !defined(OS_ANDROID)
+#endif  // defined(__GLIBC__)
diff --git chromium-75.0.3770.100/net/dns/host_resolver_manager.cc chromium-75.0.3770.100/net/dns/host_resolver_manager.cc
index 3c75dde..c748d94 100644
--- chromium-75.0.3770.100/net/dns/host_resolver_manager.cc
+++ chromium-75.0.3770.100/net/dns/host_resolver_manager.cc
@@ -2266,8 +2266,7 @@ HostResolverManager::HostResolverManager(const Options& options,
   NetworkChangeNotifier::AddIPAddressObserver(this);
   NetworkChangeNotifier::AddConnectionTypeObserver(this);
   NetworkChangeNotifier::AddDNSObserver(this);
-#if defined(OS_POSIX) && !defined(OS_MACOSX) && !defined(OS_OPENBSD) && \
-    !defined(OS_ANDROID)
+#if defined(__GLIBC__)
   EnsureDnsReloaderInit();
 #endif
 
diff --git chromium-75.0.3770.100/net/dns/host_resolver_proc.cc chromium-75.0.3770.100/net/dns/host_resolver_proc.cc
index 90d9958..71d9b4a 100644
--- chromium-75.0.3770.100/net/dns/host_resolver_proc.cc
+++ chromium-75.0.3770.100/net/dns/host_resolver_proc.cc
@@ -197,8 +197,7 @@ int SystemHostResolverCall(const std::string& host,
   base::ScopedBlockingCall scoped_blocking_call(FROM_HERE,
                                                 base::BlockingType::WILL_BLOCK);
 
-#if defined(OS_POSIX) && !defined(OS_MACOSX) && !defined(OS_OPENBSD) && \
-    !defined(OS_ANDROID) && !defined(OS_FUCHSIA)
+#if defined(__GLIBC__)
   DnsReloaderMaybeReload();
 #endif
   int err = getaddrinfo(host.c_str(), nullptr, &hints, &ai);
