diff --git chromium-75.0.3770.100/third_party/crashpad/crashpad/util/linux/ptracer.cc chromium-75.0.3770.100/third_party/crashpad/crashpad/util/linux/ptracer.cc
index c6c9229..6731148 100644
--- chromium-75.0.3770.100/third_party/crashpad/crashpad/util/linux/ptracer.cc
+++ chromium-75.0.3770.100/third_party/crashpad/crashpad/util/linux/ptracer.cc
@@ -66,6 +66,7 @@ bool GetThreadArea32(pid_t tid,
                      const ThreadContext& context,
                      LinuxVMAddress* address,
                      bool can_log) {
+#ifdef PTRACE_GET_THREAD_AREA
   size_t index = (context.t32.xgs & 0xffff) >> 3;
   user_desc desc;
   if (ptrace(
@@ -77,6 +78,9 @@ bool GetThreadArea32(pid_t tid,
 
   *address = desc.base_addr;
   return true;
+#else
+  return false;
+#endif
 }
 
 bool GetThreadArea64(pid_t tid,
